peachjam/views/legislation.py

```diff
@@ -17,3 +22,77 @@ class LegislationListView(FilteredDocumentListView):
 class LegislationDetailView(BaseDocumentDetailView):
     model = Legislation
     template_name = "peachjam/legislation_detail.html"
+
+    def get_notices(self):
+        notices = super().get_notices()
+        repeal = self.get_repeal_info()
+        if self.object.repealed and repeal:
+            msg = "This {} was repealed on {} by <a href='{}'>{}</a>"
+            notices.append(
+                {
+                    "type": messages.ERROR,
+                    "html": format_html(
+                        msg.format(
+                            self.object.metadata_json["type_name"],
+                            repeal["date"],
+                            repeal["repealing_uri"],
+                            repeal["repealing_title"],
+                        )
+                    ),
+                }
+            )
+
+            msg = (
+                "This is the latest version of this {} as it was when it was repealed."
+            )
+            notices.append(
+                {
+                    "type": messages.INFO,
+                    "html": format_html(
+                        msg.format(
+                            self.object.metadata_json["type_name"],
+                        )
+                    ),
+                }
+            )
+
+        # fetch the points in time
+        points_in_time = self.get_points_in_time()
+        latest_expression = points_in_time[-1]
+
+        for idx, point_in_time in enumerate(points_in_time):
```

Comment:
Rather than iterating over all the points in time, you want to find the index of the current point in time in the list. That tells you the start and end dates for that expression (the end date will be "never" if this is the latest point in time).